docs/tutorial/esm-limitations.md

```diff
@@ -0,0 +1,17 @@
+# ESM Limitations
```

Comment:
docs: Before this ships to folks, I'd love to have a full "and here's ESM" guide that approaches this holistically. cc @electron/docs

Reply:
And a technical blog post might be interesting too if you have the time to do some knowledge sharing!